### PR TITLE
Enable PYTHONDEVMODE in the tests

### PR DESCRIPTION
**Pull request by marcospri**
https://docs.python.org/3/library/devmode.html#devmode

From https://docs.python.org/3/whatsnew/3.9.html#you-should-check-for-deprecationwarning-in-your-code
```
More generally, try to run your tests in the Python Development Mode
which helps to prepare your code to make it
compatible with the next Python version.
```

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -13,6 +13,7 @@ setenv =
     PYTHONUNBUFFERED = 1
     OBJC_DISABLE_INITIALIZE_FORK_SAFETY = YES
     dev: DEV = {env:DEV:true}
+    dev,tests: PYTHONDEVMODE = {env:PYTHONDEVMODE:1}
```
Comment (marcospri) [inline tox.ini:16]: Not sure what the right scope for this is. `dev` and `tests`? 

Or all of them, ie, with no scope:

`PYTHONDEVMODE = {env:PYTHONDEVMODE:1}`

**Comment by marcospri** [comment]
Probably cookie-cutter stuff but wanted to bring it up on LMS first.

File: tox.ini (hunk 1 above)
Comment (marcospri) [reply tox.ini:16]: In `make dev` we get a bunch of


```web-https (stderr)   | /home/marcos/hypo/lms/.tox/dev/lib/python3.8/site-packages/jsonschema/_validators.py:321: ResourceWarning: unclosed <ssl.SSLSocket fd=11, family=AddressFamily.AF_INET, type=SocketKind.SOCK_STREAM, proto=6, laddr=('192.168.1.22', 55836), raddr=('34.232.82.173', 443)>
```

with this.

Maybe we should use it only for the tests then?

File: tox.ini [hunk 2]
```diff
@@ -20,6 +20,7 @@ setenv =
     dev: ALEMBIC_CONFIG = {env:ALEMBIC_CONFIG:conf/alembic.ini}
     dev: DATABASE_URL = {env:DATABASE_URL:postgresql://postgres@localhost:5433/postgres}
     tests: DATABASE_URL = {env:UNITTESTS_DATABASE_URL:postgresql://postgres@localhost:5433/lms_tests}
+    tests: PYTHONDEVMODE = {env:PYTHONDEVMODE:1}
```
Comment (seanh) [inline tox.ini:23]: I guess we could enable it in dev and functests as well?

```suggestion
    dev,tests,functests: PYTHONDEVMODE = {env:PYTHONDEVMODE:1}
```

**Review by seanh — APPROVED** [review]

File: tox.ini (hunk 2 above)
Comment (seanh) [reply tox.ini:23]: Oh, and of course this will have to be backported into the cookiecutter's repo

**Pull request merged**